plugins/notifications-backend/src/service/types.ts

```diff
@@ -36,6 +36,7 @@ export type NotificationsFilter = {
   createdAfter?: Date;
   messageScope?: 'all' | 'user' | 'system';
   user?: string;
+  read?: string; // '0' for unread. '1' for read. undefined for both
```

Comment:
We use string for numeric value which is boolean in fact.

Reply:
the value comes from JSON in HTTP body. so like it or not it is a string. now the question is how do we receive the value from router/express and what does typescript/javascript do for us.
The reality is that they do nothing for us. I get the string from the query object. it can actually be a string array and other types. so i have to put code in router to handle it. now.... typescript does not guarantee that my boolean function var will receive something other than true/false. Si i have to put code in handler as well.
Therefore, my approach is: pass the query object as filter object and do all the handling in handler. in order to compile that the members of filter have to be optional.
Now....since the conversion to boolean from string in javascript is terrible I have to declare the type string and not boolean and do the conversion myself.
Bottom line...it is a string because it is the query parameter and not a converted parameter.